pkg/runner/step_docker.go

```diff
@@ -0,0 +1,121 @@
+package runner
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/kballard/go-shellquote"
+	"github.com/nektos/act/pkg/common"
+	"github.com/nektos/act/pkg/container"
+	"github.com/nektos/act/pkg/model"
+)
+
+type stepDocker struct {
+	Step       *model.Step
+	RunContext *RunContext
+	env        map[string]string
+}
+
+func (sd *stepDocker) pre() common.Executor {
+	return func(ctx context.Context) error {
+		return nil
+	}
+}
+
+func (sd *stepDocker) main() common.Executor {
+	sd.env = map[string]string{}
+
+	return runStepExecutor(sd, sd.runUsesContainer())
+}
+
+func (sd *stepDocker) post() common.Executor {
+	return func(ctx context.Context) error {
+		return nil
+	}
+}
+
+func (sd *stepDocker) getRunContext() *RunContext {
+	return sd.RunContext
+}
+
+func (sd *stepDocker) getStepModel() *model.Step {
+	return sd.Step
+}
+
+func (sd *stepDocker) getEnv() *map[string]string {
+	return &sd.env
+}
+
+func (sd *stepDocker) runUsesContainer() common.Executor {
+	rc := sd.RunContext
+	step := sd.Step
+
+	return func(ctx context.Context) error {
+		image := strings.TrimPrefix(step.Uses, "docker://")
+		eval := sd.RunContext.NewExpressionEvaluator()
```

Comment:
Shouldn't you use `rc` here?
```suggestion
		eval := rc.NewExpressionEvaluator()
```